### ISSUE TITLE
Improve RAJA::forallN documentation

### ISSUE THREAD

**Issue by tepperly** [LLNL/RAJA#104]
It would be really helpful to have more documentation about the different policies for RAJA::forallN.  What's the semantics of z, y, and x in cuda_block_z_exec, cuda_threadblock_y_exec, and cuda_threadblock_x_exec? Is one preferable for the stride-1 dimension in an iteration?

For OpenMP policies, can I get collapse without getting nowait?  What are all the available options, and what do they mean?

**Comment by DavidPoliakoff** [comment]
@tepperly : thanks for a really nice point, this stuff isn't terribly well documented (I've made mistakes myself trying to use them). It looks like @davidbeckingsale already assigned some folks to work on it.

I'm mostly commenting to remind us (the devs) that @trws put up #63 a while ago and it has mostly just sat there (though we've poked at it without mentioning it by name). I think that after testing gets settled it might be wise to spend some time really hammering on that issue, I think Tom could have made the same comment on any number of other parts of the code.
